Tasks/AzureRmWebAppDeployment/MSDeployUtility.ts

```diff
@@ -23,26 +24,43 @@ export function fileExists(path) {
 
 export function getMSDeployCmdArgs(packageFile: string, webAppNameForMSDeployCmd: string, publishingProfile,
                              removeAdditionalFilesFlag: boolean, excludeFilesFromAppDataFlag: boolean, takeAppOfflineFlag: boolean,
-                             virtualApplication: string, setParametersFile: string, additionalArguments: string) : string {
+                             virtualApplication: string, setParametersFile: string, additionalArguments: string, isParamFilePresentInPacakge: boolean, isFolderBasedDeployment:boolean) : string {
 
-    var msDeployCmdArgs = ' -verb:sync';
-    msDeployCmdArgs += ' -source:package=\'' + packageFile + '\'';
-    msDeployCmdArgs += ' -dest:auto,ComputerName=https://' + publishingProfile.publishUrl + '/msdeploy.axd?site=' + webAppNameForMSDeployCmd + ',';
-    msDeployCmdArgs += 'userName=' + publishingProfile.userName + ',Password=' + publishingProfile.userPWD + ',AuthType=Basic';
+    var msDeployCmdArgs: string = " -verb:sync";
 
-    if (setParametersFile) {
-        msDeployCmdArgs += ' -setParamFile=' + setParametersFile;
-    }
+    var webApplicationDeploymentPath = ( virtualApplication ) ? webAppNameForMSDeployCmd+"/"+virtualApplication : webAppNameForMSDeployCmd ;
     
-    if (virtualApplication) {
-        msDeployCmdArgs += ' -setParam:name=\'IIS Web Application Name\',value=\'' + webAppNameForMSDeployCmd + '/' + virtualApplication + '\'';
+    if( isFolderBasedDeployment ){
+
+        msDeployCmdArgs += " -source:IisApp='"+ packageFile + "'";
+        msDeployCmdArgs += " -dest:iisApp='" + webApplicationDeploymentPath + "',";
+
+    } else {
+        
+        msDeployCmdArgs += " -source:package='"+ packageFile + "'";
+
+        if( isParamFilePresentInPacakge ){
+            msDeployCmdArgs += " -dest:auto,";           
+        } else {
+            msDeployCmdArgs += " -dest:contentPath='"+ webApplicationDeploymentPath +"',";
```

Comment:
If param file is present or not present can you try if content path works for both